core/app/models/spree/line_item.rb

```diff
@@ -106,16 +106,19 @@ def insufficient_stock?
       !sufficient_stock?
     end
 
-    # Sets the options on the line item according to the order's currency or
-    # one passed in.
+    # Sets options on the line item.
+    #
+    # The option can be arbitrary attributes on the LineItem. If no price is given in the options,
+    # this will call the legacy `PriceModifier` line item pricer.
     #
     # @param options [Hash] options for this line item
     def options=(options = {})
       return unless options.present?
 
-      self.price = variant.price_in(currency).amount +
-                   variant.price_modifier_amount_in(currency, options)
-
+      # There's no need to call a pricer if we'll set the price directly.
+      unless options[:price].is_a?(Numeric)
```

Comment:
There's specs for exactly that case which I don't want to see raising the deprecation warning for the `price_modifier` system. That was the motivation behind this check. 

Your suggestion to not check the type is good, will amend later today.